db/schema_tables.cc

```diff
@@ -102,6 +102,13 @@ namespace {
             props.enable_schema_commitlog();
```

Comment:
> raft: fast tombstone GC for group0-managed tables

Please split this huge commit into multiple logical commits for easier review.

The commit message mostly describes the different logical units:

- in one commit, add the gossip state; update it on every group 0 apply. The state is not yet used for anything in that commit
- in one commit, introduce the concept of group 0 table and migrate the selected subset of tables
- in one commit, react to the state_id gossip changes, updating "repair time" for every group 0 table
- `ensure_group0_schema` looks like additional verification that belongs to another commit as well